File: Programação/Facul/aula pablo/java/portfolio3/veiculos/CarroVenda.java
package veiculos;

import java.math.BigDecimal;
import java.text.NumberFormat;
import java.util.Locale;

public class CarroVenda extends Veiculo {
    protected BigDecimal precoVenda;

    public BigDecimal getPrecoVenda() {
        return precoVenda;
    }

    public void setPrecoVenda(BigDecimal precoVenda) {
        this.precoVenda = precoVenda;
    }

    public String obterFicha() {
    return exibirDetalhes();
}

    @Override
    protected String exibirDetalhes() {
        NumberFormat formatter = NumberFormat.getCurrencyInstance(Locale.of("pt", "BR"));
        return String.format("%s%nPreço de Venda: %s",
                super.exibirDetalhes(),
                formatter.format(precoVenda));
    }
}

File: Programação/Facul/aula pablo/c#/portfolio_3/veiculos/CarroVenda.cs
using System.Globalization;

namespace Veiculos;

public class CarroVenda : Veiculo
{
    public decimal PrecoVenda { get; set; }

    public string ObterFicha()
    {
        return ExibirDetalhes();
    }

    protected override string ExibirDetalhes()
    {
        var culturaPtBr = new CultureInfo("pt-BR");
        return $"{base.ExibirDetalhes()}\nPreco de Venda: {PrecoVenda.ToString("C", culturaPtBr)}";
    }
}